src/statistics.cpp

```diff
@@ -376,6 +436,18 @@ void stats::read(const config& cfg)
 	if (const config &c = cfg.child("defends")) {
 		defends = read_battle_result_map(c);
 	}
+	if (const config &c = cfg.child("by_cth_inflicted")) {
+		by_cth_inflicted = read_by_cth_map(c);
+	}
+	if (const config &c = cfg.child("by_cth_taken")) {
+		by_cth_taken = read_by_cth_map(c);
+	}
+	if (const config &c = cfg.child("turn_by_cth_inflicted")) {
+		turn_by_cth_inflicted = read_by_cth_map(c);
+	}
+	if (const config &c = cfg.child("turn_by_cth_taken")) {
+		turn_by_cth_taken = read_by_cth_map(c);
+	}
```

Comment:
I think we can remove `by_cth_inflicted`/`by_cth_taken` from serialization, and instead derive them from the `attacks` and `defends` members, which are read a few lines above:

https://github.com/wesnoth/wesnoth/blob/635401a307c29ce79b29034665e8be9b399755d3/src/statistics.hpp#L41-L50

With this change we'll be able to show the hit statistics for old savefiles too.

The `turn_by_cth_*` members should still be serialized.

<details>

<summary>
Savefile excerpt
</summary>

```
			[attacks]
				[sequence]
					1="s1001"
					2="s00"
					3="s11,s1,s10"
					7="s01"
					_num=40
				[/sequence]
				[sequence]
					2="s0101,s10"
					4="s11"
					5="s00,s01"
					_num=50
				[/sequence]
				[sequence]
					1="s001,s0010,s0100,s0110,s0111,s1011,s1100,s1101,s111"
					2="s10"
					4="s00"
					7="s11,s01"
					_num=60
				[/sequence]
				[sequence]
					1="s00,s11"
					_num=80
				[/sequence]
			[/attacks]
			[defends]
				[sequence]
					12="s"
					_num=0
				[/sequence]
				[sequence]
					1="s00"
					_num=40
				[/sequence]
				[sequence]
					1="s10"
					_num=50
				[/sequence]
				[sequence]
					1="s1,s10,s101"
					3="s00"
					5="s"
					9="s11"
					11="s01"
					_num=60
				[/sequence]
				[sequence]
					1="s01,s11"
					_num=80
				[/sequence]
			[/defends]
```

The first block in the `defends` section may need special casing, but other than that it looks straightforward enough.

</details>